lib/parsers/moov_parser/decoder.rb

```diff
@@ -4,6 +4,7 @@
 # To know more about Atoms: https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/QTFFChap2/qtff2.html
 class FormatParser::MOOVParser::Decoder
   include FormatParser::IOUtils
+  include FormatParser::EXIFParser
```

Comment:
Feels odd to add an image specific dependency (exif parser) to a moov specific decoder. Similar to some other comments I added it feel a bit like there issome core common functionality regarding atoms and then 2 quite specific uses of this (one the existing moov logic and the other the new image related logic).

Would it be possible to have a common AtomDecoder (usable anywhere) then this moov decoder that wraps the atomdecoder with moov specific logic and a separate CR3 decoder which wraps the AtomDecoder with cr3 specific logic?

Reply:
Yeah I like that as an idea, I'll give it a go!